### PR TITLE
Added linter for csharp

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Added superLinter to verify markdown files and .cs files using editorConfig file 
- Removed markdown lint stage as it is no longer needed 
- Linter will not verify existing issues as VALIDATE_ALL_CODEBASE is set to false.
- editorconfig file contains configuration for IDE and gives suggestions and warning related to linting to make sure user follows the rules.

### CODE REVIEW

File: .github/workflows/csharp.yml [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          # Full git history is needed to get a proper list of changed files
+          fetch-depth: 0
+      - name: Lint Code Base
+        uses: github/super-linter@v4
+        env:
+          VALIDATE_ALL_CODEBASE: false
+          DEFAULT_BRANCH: master
```
Comment (msohailhussain) [inline .github/workflows/csharp.yml:24]: does this mean it only checks for master branch?

File: .github/workflows/csharp.yml [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (msohailhussain) [inline .github/workflows/csharp.yml:25]: Why github token is needed?

File: .github/workflows/csharp.yml (hunk 1 above)
Comment (mnoman09) [reply .github/workflows/csharp.yml:24]: yes It will compare and check linting of only the files which were modified and compares with master.

**Review by msohailhussain — APPROVED** [review]
looks cool.

**Pull request merged**